packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -284,12 +284,54 @@ class NestedScrollView extends StatefulWidget {
   }
 
   @override
-  _NestedScrollViewState createState() => _NestedScrollViewState();
+  NestedScrollViewState createState() => NestedScrollViewState();
 }
 
-class _NestedScrollViewState extends State<NestedScrollView> {
+/// This is the [State] for [NestedScrollView]. It allows you to obtain the scroll
+/// position of the children of your nested scroll view.
+///
+/// You can access the inner scroll controller via [innerController] and the outer
+/// controller with [outerController], however, notice that the outer scroll position
+/// can also be obtained using [NestedScrollView.controller].
+///
+/// If you want to access the inner scroll controller of your nested scroll view,
+/// you can get its [NestedScrollViewState] by supplying a `GlobalKey<NestedScrollViewState>`
+/// to your [NestedScrollView] widget (the `key` parameter).
+///
+/// {@tool sample}
+///
+/// You would access the [NestedScrollViewState] using a [GlobalKey].
+///
+/// ```dart
+/// @override
+/// Widget build(BuildContext context) {
+///   // You probably want to declare your GlobalKey outside of your build method.
+///   final GlobalKey<NestedScrollViewState> globalKey = GlobalKey();
+///
+///   return NestedScrollView(
+///     key: globalKey,
+///     headerSliverBuilder: (context, _) => const [SliverAppBar()],
+///     body: const CustomScrollView(
+///       // ...
+///     ),
+///   );
+/// }
+/// ```
+///
+/// Now, you can access the inner scroll controller using `globalKey.currentState.innerController`.
+/// {@end-tool}
+///
+/// Alternatively, you could also visit child elements of your parent widget to obtain
+/// the state if you know exactly that the widget is a direct descendant.
+class NestedScrollViewState extends State<NestedScrollView> {
   final SliverOverlapAbsorberHandle _absorberHandle = SliverOverlapAbsorberHandle();
 
+  /// This controls the scroll of the scroll view in [NestedScrollView.body].
+  ScrollController get innerController => _coordinator._innerController;
+
+  /// This controls the scroll of the scroll view in [NestedScrollView.headerSliverBuilder].
```

Comment:
Maybe: The [ScrollController] used by the [ScrollView] containing the [Slivers] from [NestedScrollView.headerSliverBuilder].

Same additional documentation and see also section should be added here.